python_ta/cfg/visitor.py

```diff
@@ -1,18 +1,25 @@
 import astroid
 from astroid.node_classes import NodeNG
 from .graph import ControlFlowGraph, CFGBlock
+from typing import List, Tuple
 
 
 class CFGVisitor:
     """An astroid visitor that creates a control flow graph for a given Python module.
+
+    Private Attributes:
+    _scope: A stack of the scopes the visitor is currently in. The top element refers
+            to the current scope. (scope_type [for, while],  start_block, end_block)
     """
     cfg: ControlFlowGraph
     _current_block: CFGBlock
+    _scope: List[Tuple[str, CFGBlock, CFGBlock]]
```

Comment:
Names are hard. Use the name `_control_boundaries` for now.

The idea of using a stack is excellent, but I want you to change the inner type as follows:
1. The first element of the tuple should be a `NodeNG`---that is, set it to the be node itself, not a name for the node.
2. The second element should be a dictionary mapping *node class names* to `CFGBlock`s, which obeys the intuitive rule "when the current statement has name `s`, the current block is connected to the dictionary's value corresponding to `s`. For example, you'd store a `While` node's "scope" as something like `(node, {'Break': after_while_block, 'Continue': test_block})`. 

Note that you can access the name of an object's class by doing `type(x).__name__`.